### PR TITLE
add support for aws ipi private sdn

### PR DESCRIPTION
**Pull request by geliu2016**
@gpei @zhaozhanqi could you please take view? thanks

### CODE REVIEW

**Comment by zhaozhanqi** [comment]
@geliu2016 I remembered openshift-sdn is not supported for installation from 4.15.  what's the workflow for?

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.15.yaml [hunk 1]
```diff
@@ -79,15 +79,15 @@ tests:
     test:
     - ref: cucushift-installer-check-cluster-health
     workflow: cucushift-installer-rehearse-aws-ipi-shared-phz
-- as: aws-ipi-shared-phz-sts-f14
+- as: aws-ipi-sdn-phz-f14
```
[the remark this answers is not in the document]
Comment (geliu2016) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__installation-nightly-4.15.yaml:82]: Updated.

**Comment by zhaozhanqi** [comment]
> > @geliu2016 I remembered openshift-sdn is not supported for installation from 4.15. what's the workflow for?
> 
> @zhaozhanqi the workflow is designed for disasterrecovery test. why 4.15 don't support sdn? which ocp version support openshift-sdn?

from 4.15, user cannot setup new cluster with openshift-sdn plugin but still can work for upgrade users.  we are expecting user migrate sdn to ovn.  I mean ovn is the future. and openshift-sdn will not be supported in 4.17 if I remember correctly.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml [hunk 2]
```diff
@@ -643,7 +643,7 @@ tests:
     test:
     - ref: openshift-extended-test
     workflow: cucushift-installer-rehearse-aws-ipi-ovn-winc
-- as: aws-ipi-private-fips-f28-disasterrecovery
+- as: aws-ipi-private-fips-sdn-f28-disasterrecovery
```
Comment (zhaozhanqi) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:646]: @geliu2016 so here you want to update the default network (ovn) to sdn?  is there another testing for ovn?

Comment (zhaozhanqi) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:646]: for sdn from 4.15 and 4.16,  you may need to add one env `OVERRIDE_OPENSHIFT_SDN_DEPRECATION: "true"`

Comment (geliu2016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:646]: @zhaozhanqi as I sync with gpei, this parameter have been enabled in https://github.com/openshift/release/blob/master/ci-operator/config/openshift/sdn/openshift-sdn-release-4.15.yaml#L95, is it exact?

Comment (geliu2016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:646]: > @geliu2016 so here you want to update the default network (ovn) to sdn? is there another testing for ovn?

the default installation is ovn, exact? I just want to add coverage for sdn

Comment (zhaozhanqi) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:646]: > @zhaozhanqi as I sync with gpei, this parameter have been enabled in https://github.com/openshift/release/blob/master/ci-operator/config/openshift/sdn/openshift-sdn-release-4.15.yaml#L95, is it exact?

it's enables in developer's job.  however you need also add this parameter in your job profile

Comment (zhaozhanqi) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:646]: > > @geliu2016 so here you want to update the default network (ovn) to sdn? is there another testing for ovn?
> 
> the default installation is ovn, exact? I just want to add coverage for sdn

Yes, default is OVN.  ok for me if this is for sdn

Comment (geliu2016) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml:646]: I have many dr profiles, most of them cover ovn, and this one cover sdn.

**Comment by geliu2016** [comment]
> > > @geliu2016 I remembered openshift-sdn is not supported for installation from 4.15. what's the workflow for?
> > 
> > 
> > @zhaozhanqi the workflow is designed for disasterrecovery test. why 4.15 don't support sdn? which ocp version support openshift-sdn?
> 
> from 4.15, user cannot setup new cluster with openshift-sdn plugin but still can work for upgrade users. we are expecting user migrate sdn to ovn. I mean ovn is the future. and openshift-sdn will not be supported in 4.17 if I remember correctly.

@zhaozhanqi so you means that 4.15 can't installed with sdn mode? but why this dev's profile config sdn for 4.15: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/sdn/openshift-sdn-release-4.15.yaml#L95

**Comment by zhaozhanqi** [comment]
> > > > @geliu2016 I remembered openshift-sdn is not supported for installation from 4.15. what's the workflow for?
> > > 
> > > 
> > > @zhaozhanqi the workflow is designed for disasterrecovery test. why 4.15 don't support sdn? which ocp version support openshift-sdn?
> > 
> > 
> > from 4.15, user cannot setup new cluster with openshift-sdn plugin but still can work for upgrade users. we are expecting user migrate sdn to ovn. I mean ovn is the future. and openshift-sdn will not be supported in 4.17 if I remember correctly.
> 
> @zhaozhanqi so you means that 4.15 can't installed with sdn mode? but why this dev's profile config sdn for 4.15: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/sdn/openshift-sdn-release-4.15.yaml#L95

hack way you need OVERRIDE_OPENSHIFT_SDN_DEPRECATION: "true"

**Comment by geliu2016** [comment]
> > > > > @geliu2016 I remembered openshift-sdn is not supported for installation from 4.15. what's the workflow for?
> > > > 
> > > > 
> > > > @zhaozhanqi the workflow is designed for disasterrecovery test. why 4.15 don't support sdn? which ocp version support openshift-sdn?
> > > 
> > > 
> > > from 4.15, user cannot setup new cluster with openshift-sdn plugin but still can work for upgrade users. we are expecting user migrate sdn to ovn. I mean ovn is the future. and openshift-sdn will not be supported in 4.17 if I remember correctly.
> > 
> > 
> > @zhaozhanqi so you means that 4.15 can't installed with sdn mode? but why this dev's profile config sdn for 4.15: https://github.com/openshift/release/blob/master/ci-operator/config/openshift/sdn/openshift-sdn-release-4.15.yaml#L95
> 
> hack way you need OVERRIDE_OPENSHIFT_SDN_DEPRECATION: "true"

@zhaozhanqi ok, I see, and I have updated this parameter in profile, do you have more comment? thanks

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-private-fips-sdn-f28-disasterrecovery

**Comment by geliu2016** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-private-fips-sdn-f28-disasterrecovery

**Comment by geliu2016** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-private-fips-sdn-f28-disasterrecovery

**Comment by geliu2016** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-private-fips-sdn-f28-disasterrecovery

**Comment by geliu2016** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.15-amd64-nightly-aws-ipi-private-fips-sdn-f28-disasterrecovery

**Comment by geliu2016** [comment]
@gpei @kuiwang02 Test job passed, could you please take a view?

**Comment by kuiwang02** [comment]
@geliu2016 please request qe-productivity team to review it.

**Comment by geliu2016** [comment]
@sandeepknd @zhaozhanqi could you pls take a view? thanks

**Comment by zhaozhanqi** [comment]
> @sandeepknd @zhaozhanqi could you pls take a view? thanks

ok for me,  FYI,  sdn will be deprecated in 4.17 version, so please remember to remove this job from 4.17.

**Comment by geliu2016** [comment]
> > @sandeepknd @zhaozhanqi could you pls take a view? thanks
> 
> ok for me, FYI, sdn will be deprecated in 4.17 version, so please remember to remove this job from 4.17.

cc: @kuiwang02

**Comment by liangxia** [comment]
@geliu2016 There are conflicts, would you mind to rebase ?

File: ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sdn/OWNERS [hunk 3]
```diff
@@ -1,14 +1,18 @@
 approvers:
 - jianlinliu
+- Amoghrd
 - yunjiang29
-- gpei
+- mgahagan73
+- MayXuQQ
```
Comment (liangxia) [inline ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sdn/OWNERS:6]: Are those expected changes in this PR ?

Comment (geliu2016) [reply ci-operator/step-registry/cucushift/installer/rehearse/aws/ipi/sdn/OWNERS:6]: I just feel confuse why this file be changed, I will rebase it.

**Comment by geliu2016** [comment]
all test passed, cc: @liangxia

**Pull request merged**